tls/s2n_x509_validator.c

```diff
@@ -457,99 +453,54 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_stapled_ocsp_response(
             break;
         }
     }
-
-    if (!issuer) {
-        goto clean_up;
-    }
+    RESULT_ENSURE(issuer != NULL, S2N_ERR_CERT_UNTRUSTED);
 
     /* Important: this checks that the stapled ocsp response CAN be verified, not that it has been verified. */
     const int ocsp_verify_res = OCSP_basic_verify(basic_response, cert_chain, validator->trust_store->trust_store, 0);
-
-    /* OCSP_basic_verify() returns 1 on success, 0 on error, or -1 on fatal error such as malloc failure. */
-    if (ocsp_verify_res != _OSSL_SUCCESS) {
-        ret_val = ocsp_verify_res == 0 ? S2N_CERT_ERR_UNTRUSTED : S2N_CERT_ERR_INTERNAL_ERROR;
-        goto clean_up;
-    }
+    RESULT_GUARD_OSSL(ocsp_verify_res, S2N_ERR_CERT_UNTRUSTED);
 
     /* do the crypto checks on the response.*/
     int status = 0;
     int reason = 0;
 
     /* sha1 is the only supported OCSP digest */
     OCSP_CERTID *cert_id = OCSP_cert_to_id(EVP_sha1(), subject, issuer);
-
-    if (!cert_id) {
-        goto clean_up;
-    }
+    RESULT_GUARD_PTR(cert_id);
 
     ASN1_GENERALIZEDTIME *revtime, *thisupd, *nextupd;
     /* Actual verification of the response */
     const int ocsp_resp_find_status_res = OCSP_resp_find_status(basic_response, cert_id, &status, &reason, &revtime, &thisupd, &nextupd);
     OCSP_CERTID_free(cert_id);
-
-    if (!ocsp_resp_find_status_res) {
-        ret_val = S2N_CERT_ERR_UNTRUSTED;
-        goto clean_up;
-    }
+    RESULT_GUARD_OSSL(ocsp_resp_find_status_res, S2N_ERR_CERT_UNTRUSTED);
 
     uint64_t this_update = 0;
-    s2n_result thisupd_result = s2n_asn1_time_to_nano_since_epoch_ticks((const char *) thisupd->data,
-                                                                  (uint32_t) thisupd->length, &this_update);
+    RESULT_GUARD(s2n_asn1_time_to_nano_since_epoch_ticks((const char *) thisupd->data,
+            (uint32_t) thisupd->length, &this_update));
 
     uint64_t next_update = 0;
-    s2n_result nextupd_result = S2N_RESULT_OK;
     if (nextupd) {
-        nextupd_result = s2n_asn1_time_to_nano_since_epoch_ticks((const char *) nextupd->data,
-                                                                  (uint32_t) nextupd->length, &next_update);
+        RESULT_GUARD(s2n_asn1_time_to_nano_since_epoch_ticks((const char *) nextupd->data,
+                (uint32_t) nextupd->length, &next_update));
     } else {
         next_update = this_update + DEFAULT_OCSP_NEXT_UPDATE_PERIOD;
     }
 
     uint64_t current_time = 0;
     const int current_time_err = conn->config->wall_clock(conn->config->sys_clock_ctx, &current_time);
+    RESULT_GUARD_POSIX(current_time_err);
```

Comment:
No need to assign the result, right?

```suggestion
    RESULT_GUARD_POSIX(conn->config->wall_clock(conn->config->sys_clock_ctx, &current_time));
```

Reply:
Yes, true. Committed! 886ddedb44bf04fccc218d70163ba58dd3ead800